### PR TITLE
TST: split up pandas/tests/indexing/test_multiindex.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [n/a ] follow on from  #23797
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ n/a] whatsnew entry

only whole tests moved, tests have not been split up. this and the removal of classes is left for a follow-on PR.

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+        # new_df = pd.concat([f(name, df2) for name, df2 in grp], axis=1).T
+
+        # we are actually operating on a copy here
+        # but in this case, that's ok
+        for name, df2 in grp:
+            new_vals = np.arange(df2.shape[0])
+            with catch_warnings(record=True):
+                df.ix[name, 'new_col'] = new_vals
+
+    def test_multiindex_label_slicing_with_negative_step(self):
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_multiindex.py]: move to slicing

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 2]
```diff
@@ -0,0 +1,907 @@
...
+        result = df_slice.loc['a']
+        expected = DataFrame(list(range(30, 71)),
+                             columns=['Test'], index=range(30, 71))
+        tm.assert_frame_equal(result, expected)
+        result = df_slice.loc['d']
+        expected = DataFrame(list(range(1530, 1571)),
+                             columns=['Test'], index=range(30, 71))
+        tm.assert_frame_equal(result, expected)
+
+    def test_multiindex_symmetric_difference(self):
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_multiindex.py]: can you move to (new file set_ops)

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 3]
```diff
@@ -0,0 +1,907 @@
...
+            self, multiindex_year_month_day_dataframe_random_data):
+        # GH #1013
+        ymd = multiindex_year_month_day_dataframe_random_data
+        df = ymd[:5]
+
+        result = df.loc[(2000, 1, 6), ['A', 'B', 'C']]
+        expected = df.loc[2000, 1, 6][['A', 'B', 'C']]
+        tm.assert_series_equal(result, expected)
+
+    def test_get_loc_single_level(self, single_level_multiindex):
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_multiindex.py]: there are some loc tests in here

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 4]
```diff
@@ -0,0 +1,907 @@
...
+
+        result = df['bar']
+        result2 = df.loc[:, 'bar']
+
+        expected = df.reindex(columns=df.columns[3:5])
+        expected.columns = expected.columns.droplevel(0)
+        tm.assert_frame_equal(result, expected)
+        tm.assert_frame_equal(result, result2)
+
+    def test_getitem_setitem_slice_integers(self):
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_multiindex.py]: can you move pure getitem tests to a new file, and pure setitem to a newfile as well, if they are both getitem & setitem, put in the setitem one

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 5]
```diff
@@ -0,0 +1,907 @@
...
+    def test_getitem_lowerdim_corner(self, multiindex_dataframe_random_data):
+        frame = multiindex_dataframe_random_data
+        pytest.raises(KeyError, frame.loc.__getitem__,
+                      (('bar', 'three'), 'B'))
+
+        # in theory should be inserting in a sorted space????
+        frame.loc[('bar', 'three'), 'B'] = 0
+        assert frame.sort_index().loc[('bar', 'three'), 'B'] == 0
+
+    def test_int_series_slicing(
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_multiindex.py]: to slicing

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 6]
```diff
@@ -0,0 +1,907 @@
...
+        result = df['a']
+        expected = df['a', '', ''].rename('a')
+        tm.assert_series_equal(result, expected)
+
+        result = df['routine1', 'result1']
+        expected = df['routine1', 'result1', '']
+        expected = expected.rename(('routine1', 'result1'))
+        tm.assert_series_equal(result, expected)
+
+    def test_mixed_depth_insert(self):
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_multiindex.py]: insert, difference to set_ops

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, some small splitting. ping on green.

**Comment by simonjayhawkins** [comment]
@jreback green!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**